_posts/2020-11-07-for-loops-how-to.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title:  "How to be successful with for loops"
+date:   2020-11-07 12:54:04 -0700
+categories: jekyll update
+---
+In this post you will learn about the basics of creating for loops and why they
+are important as a bioinformatics tool.
+
+# What to know beforehand
+In order to utilize for loops it is best to have a working function. The main
+advantage of being able to use a function is the fact that it makes your code
+incredibly flexible. With a properly working function all you would need to do is
+have an input and the code will work to manipulate your input data and produce a workable output. Without the function, your code will only work for a specific input and only produce a singular output.  While still useful making the code more flexible allows you to perform
+the same action on multiple different potential inputs. This is an example of a basic function
+
+```
+square_this <- function(x) {
+  if (!is.numeric(x)) {
+    stop("ERROR: You did not use a numeric value.")
+  }
+  x^2
+}
+```
+This function is designed to take a numeric input (if it is a non numeric input the function will not run) and add two to it. It then stores this function into the variable "square_this" in order to call the function for use at a later time.  
+
+# What is a for loop?
+Using a function makes your code more flexible. But say you had numerous different inputs you wanted to run. Running the function for each input one at a time can be tedious. If you had hundreds of different inputs to run inputting each and every one would be impractical. This is where for loops come in.
```

Comment:
I think it would be really helpful to add a bioinformatics specific reason here! Something like 

If you had hundreds of different inputs, for example sequences in a fasta file, etc.